myria/cluster/scripts/cli.py

```diff
@@ -1141,7 +1141,20 @@ def run():
     help="Number of Myria workers per cluster node [default: %d]" % INSTANCE_TYPE_CONFIGS[DEFAULTS['instance_type']].workers_per_node)
 @click.option('--cluster-log-level', show_default=True,
     type=click.Choice(LOG_LEVELS), default=DEFAULTS['cluster_log_level'])
+@click.option('--perfenforce', is_flag=True, help="Enable PerfEnforce (will override default cluster configuration)")
 def create_cluster(cluster_name, **kwargs):
+    # If perfenforce is enabled, override the cluster configuration
+    if kwargs['perfenforce']:
+        click.echo("Adjusting cluster options for PerfEnforce...")
```

Comment:
I'm running into the same issue now and am getting frustrated enough that I'm ready to fork Click to use declaration order for option precedence. I'm spent a ridiculous amount of time trying to hack around the invocation order precedence, and the code is needlessly complex and fragile as a result.